File: filesystem/src/grammar/tests/validation_data/cpp/validator_incomplete_syntax.cpp
// C++ validation file with incomplete syntax to test parser robustness

#include <iostream>
#include <string>
#include <vector>
#include <memory>

// Incomplete class declaration
class IncompleteClass {
public:
    IncompleteClass(const std::string& name, int value
        // Missing closing parenthesis
    
    // Incomplete method
    void process(const std::vector<int>& data
        // Missing closing parenthesis and method body
    
private:
    std::string name_;
    int value_;
    // Missing closing brace

// Incomplete namespace
namespace incomplete {
    // Incomplete function
    void process_data(int x, int y
        // Missing closing parenthesis and function body
    
    // Incomplete enum
    enum class Status {
        Active,
        Pending,
        // Missing closing brace
    
    // Missing closing brace

// Incomplete template class
template <typename T, typename U
    // Missing closing angle bracket
class TemplateClass {
public:
    TemplateClass(T value) : value_(value) {
    
    T getValue() const {
        return value_;
    
private:
    T value_;
    // Missing closing brace

// Incomplete nested template
template <template <typename> class Container, typename T>
class NestedTemplate {
    Container<T data_;
    // Missing closing angle bracket and semicolon
    
    // Missing closing brace

// Incomplete function template
template <typename T
    // Missing closing angle bracket
T max_value(T a, T b) {
    return (a > b) ? a : b;
    // Missing closing brace

// Incomplete lambda expression
auto lambda = [](int x, int y
    // Missing closing parenthesis and lambda body

// Incomplete if statement
void conditional_function(int value) {
    if (value > 0
        // Missing closing parenthesis and if body
    else {
        std::cout << "Value is not positive\n";
    // Missing closing brace

// Incomplete for loop
void loop_function(const std::vector<int>& values) {
    for (size_t i = 0; i < values.size(
        // Missing closing parenthesis and loop body
    
    // Incomplete range-based for
    for (auto value : values
        // Missing closing parenthesis and loop body
    
    // Missing closing brace

// Incomplete try-catch block
void exception_function() {
    try {
        throw std::runtime_error("Error occurred");
    catch (const std::exception& e
        // Missing closing parenthesis and catch body
    
    // Missing closing brace

// Incomplete template specialization
template <>
class TemplateClass<int, double
    // Missing closing angle bracket
public:
    // Specialization implementation
    // Missing closing brace

// Incomplete structured binding
void structured_binding(const std::pair<int, std::string>& pair) {
    auto [value, name
        // Missing closing bracket and statement
    
    // Missing closing brace

// Incomplete multi-line string literal
const char* text = R"(
    This is a multi-line string
    with no closing delimiter

// Incomplete nested blocks and initializers
void nested_incomplete() {
    if (true) {
        for (int i = 0; i < 10; i++) {
            struct {
                int x;
                int y;
                // Missing closing brace
            } point = {
                1,
                // Missing closing brace and semicolon
            
            // Missing closing brace for loop
        // Missing closing brace for if
    // Missing closing brace for function 

File: filesystem/src/grammar/tests/validation_data/rs/validator_incomplete_syntax.rs
// Rust validation file with incomplete syntax to test parser robustness

// Incomplete struct definition
struct IncompleteStruct {
    name: String,
    value: u32,
    // Missing closing brace

// Incomplete enum
enum Status {
    Active,
    Pending,
    Completed(
        // Missing type and closing parenthesis
    Failed {
        // Missing field and closing brace

// Incomplete function declaration
fn process_data(data: Vec<u32>
    // Missing closing parenthesis and function body

// Incomplete match statement
fn incomplete_match(value: Option<u32>) -> u32 {
    match value {
        Some(val) => val,
        None => 
            // Missing expression
    // Missing closing brace

// Incomplete impl block
impl IncompleteStruct {
    fn new(name: &str) -> Self {
        Self {
            name: name.to_string(),
            value: 42,
            // Missing closing brace and parenthesis
    
    fn get_value(&self
        // Missing closing parenthesis and function body

// Incomplete trait definition
trait DataProcessor {
    fn process(&self, data: &[u8]) -> Vec<u8>;
    
    fn validate(&self, input: &str
        // Missing closing parenthesis and function body

// Incomplete use statement
use std::collections::{
    HashMap, 
    // Missing closing brace

// Incomplete macro invocation
println!(
    "Value is: {}",
    // Missing closing parenthesis

// Incomplete let statement with pattern matching
let (x, y
    // Missing closing parenthesis and expression

// Incomplete if expression
if condition {
    println!("Condition is true");
} else 
    // Missing opening brace for else block

// Incomplete for loop
for item in items
    // Missing opening brace

// Incomplete closure
let closure = |x: u32, y: u32
    // Missing closing pipe and body

// Incomplete vector declaration
let values = vec![
    1, 2, 3,
    // Missing closing bracket

// Incomplete string literal
let message = "This string has no ending quote

// Incomplete lifetime parameter
fn with_lifetime<'a>(data: &'a [u32]
    // Missing closing angle bracket and function body

// Incomplete where clause
fn generic_function<T>(value: T) -> T 
where 
    T: Clone + 
    // Missing trait bound

// Incomplete async function with nested blocks
async fn fetch_data(url: &str) -> Result<String, Error> {
    let client = Client::new();
    
    let response = client.get(url)
        .send()
        .await
        // Missing questionmark or semicolon
        
    match response {
        Ok(res) => {
            if res.status().is_success() {
                let text = res.text().await
                // Missing questionmark or semicolon
                Ok(text
                // Missing closing parenthesis
            } else {
                Err(
                // Missing error and closing parenthesis
        // Missing closing brace and Err case
} 